services/QuillLMS/client/app/bundles/Teacher/components/general_components/activity_icon_with_tooltip.jsx

```diff
@@ -4,10 +4,56 @@ import React from 'react';
 import { nonRelevantActivityClassificationIds, } from '../../../../modules/activity_classifications';
 import { requestGet, } from '../../../../modules/request/index';
 import { closedLockIcon, scheduledIcon, } from '../../../Shared/index';
-import activityFromClassificationId from '../modules/activity_from_classification_id.js';
 import ScorebookTooltip from '../modules/componentGenerators/tooltip_title/scorebook_tooltip_title';
 import gradeColor from '../modules/grade_color.js';
 
+function skillDescription(grade) {
+  if (grade == null) {
+    return 'did not complete the activity';
+  } else if (grade < 0.32) {
+    return 'rarely demonstrated the activity\'s skill';
+  } else if (grade < 0.83) {
+    return 'sometimes demonstrated the activity\'s skill';
+  } else if (grade <= 1.0) {
+    return 'frequently demonstrated the activity\'s skill';
+  }
+  return 'did not complete the activity';
+};
+
+function activityFromClassificationId(classificationId) {
+  const intClassificationId = parseInt(classificationId);
+  if (intClassificationId === 1) {
```

Comment:
Thomas's comment could also apply in this function as well (use the same dictionary lookup method to translate ID to classification name).